SopraGameAction.cpp

```diff
@@ -0,0 +1,197 @@
+#include "SopraGameAction.h"
+namespace gameController{
+    Action::Action(const std::shared_ptr<gameModel::Player> actor, const gameModel::Position target) :
+        actor(actor), target(target){}
+
+
+    Shot::Shot(const std::shared_ptr<gameModel::Player> actor, const gameModel::Position target) :
+            Action(actor, target) {}
+
+    void Shot::execute(gameModel::Environment &envi) const{
+        if (check(envi) == ActionResult::Impossible){
+            throw std::runtime_error("Action is impossible");
+        }
+
+        for(const auto &pos : getInterceptionPositions(envi)){
+            if(gameController::actionTriggered(envi.config.gameDynamicsProbs.catchQuaffle)){
+                auto player = envi.getPlayer(pos);
+                if(player.has_value()) {
+                    //@TODO
+                } else {
+                    throw std::runtime_error("No player at specified interception point");
+                }
+            }
+        }
+    }
+
+    auto Shot::successProb(const gameModel::Environment &envi) const -> double{
+
+        if (gameModel::Environment::getCell(this->target) == gameModel::Cell::OutOfBounds){
+            return 0;
+        }
+        else {
+            // @ToDo: Wahrscheinlichkeit für das Abfangen einfügen.
+            return pow(envi.config.gameDynamicsProbs.throwSuccess,
```

Comment:
```suggestion
            return std::pow(envi.config.gameDynamicsProbs.throwSuccess,
```
pow ist C und nicht überladen, std::pow ist überladen für die meisten Typen